design/trigger-backups-api-events.md

```diff
@@ -0,0 +1,35 @@
+# Trigger backups based on Kubernetes events 
+
+Triggering backups with Velero based on Kubernetes api events (Terminating, deleting, CrashLoopBackOff, etc..).
+
+## Goals
+
+- Rather than performing backups manually or by scheduling them, backups are created based on events.
+- Recovering the components of the cluster when they were deleted accidently. 
+- Blocking the delete process of a component in the cluster if it was a human intervention and trigger a backup.
+
+## Non Goals
+
+- N/A
```

Comment:
I think some non-goals here might be useful in providing constraints around the design.